src/spec-node/devContainersSpecCLI.ts

```diff
@@ -735,8 +746,9 @@ async function readConfiguration({
 			};
 		}
 
-		const needsFeaturesConfig = includeFeaturesConfig || (includeMergedConfig && (!container || !experimentalImageMetadata));
-		const featuresConfiguration = needsFeaturesConfig ? await readFeaturesConfig(params, pkg, configuration.config, extensionPath, skipFeatureAutoMapping) : undefined;
+		const additionalFeatures = additionalFeaturesJson ? jsonc.parse(additionalFeaturesJson) as Record<string, string | boolean | Record<string, string | boolean>> : undefined;
+		const needsFeaturesConfig = includeFeaturesConfig || additionalFeatures || (includeMergedConfig && (!container || !experimentalImageMetadata));
```

Comment:
The features config shouldn't be needed when the container already exists. The container will have the config from the additional features on its devcontainers.metadata label. (Fetching the features config can be expensive.) Please remove the `additionalFeatures` from the condition.

Reply:
Good point - updated